src/main/java/com/yahoo/identity/services/storage/sql/SqlAccountVulnerable.java

```diff
@@ -0,0 +1,163 @@
+package com.yahoo.identity.services.storage.sql;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+import com.yahoo.identity.IdentityException;
+import com.yahoo.identity.services.account.Account;
+import com.yahoo.identity.services.account.AccountUpdate;
+import com.yahoo.identity.services.storage.AccountModel;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.ibatis.session.SqlSession;
+import org.apache.ibatis.session.SqlSessionFactory;
+
+import java.time.Instant;
+
+import javax.annotation.Nonnull;
+import javax.ws.rs.NotAuthorizedException;
+
+public class SqlAccountVulnerable implements Account {
+
+    private static final int ABUSE_MAX_TRIES = 5;
+    private static final long ABUSE_MIN_BLOCK = 300;
+    private static final long ABUSE_MAX_BLOCK = 7200;
+    private static final long ABUSE_BLOCK_FACTOR = 2;
+    private static final int ABUSE_RESET_ZERO = 0;
+    private AccountModel account;
+    private SqlSessionFactory sqlSessionFactory;
+
+    public SqlAccountVulnerable(@Nonnull SqlSessionFactory sqlSessionFactory) throws IdentityException {
+        this.sqlSessionFactory = sqlSessionFactory;
+    }
+
+    @Nonnull
+    public void getAccount(@Nonnull String username) {
```

Comment:
`Account` should be an immutable class representing an existing account. It shouldn't have such interface.